.github/workflows/ci.yml

```diff
@@ -55,7 +56,14 @@ jobs:
           ASAN_OPTIONS: "detect_leaks=0"
         run: |
           export LD_PRELOAD=`gcc -print-file-name=libasan.so`
-          py.test -s -m "not mypy and not pyright"
+          coverage run -m pytest -s -m "not mypy and not pyright"
+          coverage xml
+          gcov -abcu `find build/ -name *.o`
```

Comment:
Here we:

- Run pytest via `coverage` to enable python coverage support
- Generate a `coverage.xml` file containing coverage for all Python source files
- Use `gcov` (which comes standard with gcc) to generate `.gcov` files for every source file in our build.